### PR TITLE
Explorer: display message when block rewards not found for interval

### PR DESCRIPTION
**Pull request by oJshua**
#### Problem
There is currently no user feedback for situations where no block rewards are found for a given interval.

#### Summary of Changes
Rather than render an empty table, display a message saying rewards were not found for the fetched interval.

Explorer-Solana (6)

Explorer-Solana (7)

### CODE REVIEW

File: explorer/src/providers/accounts/rewards.tsx [hunk 1]
```diff
@@ -111,13 +115,20 @@ async function fetchRewards(
...
       const result = await connection.getInflationReward([pubkey], epoch);
-      return result[0];
+      if (result[0] !== null) {
+        return result[0];
+      }
     } catch (error) {
-      if (cluster !== Cluster.Custom) {
+      if (
+        cluster !== Cluster.Custom &&
+        !error.message.match(/Block not available/)
```
Comment (jstarry) [inline explorer/src/providers/accounts/rewards.tsx:124]: Is this an actual response from this rpc method?

File: explorer/src/providers/accounts/rewards.tsx [hunk 2]
```diff
@@ -111,13 +115,20 @@ async function fetchRewards(
...
+      ) {
         reportError(error, { url });
       }
     }
-    return null;
+    return {
+      epoch,
+    };
```
Comment (jstarry) [inline explorer/src/providers/accounts/rewards.tsx:131]: Shouldn't the return value be different between a null response and a failed request?

File: explorer/src/providers/accounts/rewards.tsx [hunk 3]
```diff
@@ -8,15 +8,19 @@ import { reportError } from "utils/sentry";
 
 const PAGE_SIZE = 15;
 
+export type NoReward = {
+  epoch: number;
```
Comment (jstarry) [inline explorer/src/providers/accounts/rewards.tsx:12]: What's the purpose of this over using `null`? Doesn't look like `epoch` is used anywhere

**Review by jstarry — COMMENTED** [review]
@oJshua I imagine there's a known epoch on each cluster where we know rewards aren't available, can we hardcode that to disable the "Load More" button at a certain point? 

On a separate note, looks like we're thrashing with each other on the `package-lock.json`. You probably want to update your `npm` to avoid these big diffs.

File: explorer/src/providers/accounts/rewards.tsx (hunk 3 above)
Comment (oJshua) [reply explorer/src/providers/accounts/rewards.tsx]: This was used in the reconciler: https://github.com/solana-labs/solana/blob/master/explorer/src/providers/accounts/rewards.tsx#L42-L43 but I think I'll go back to using null instead.

**Comment by oJshua** [comment]
Ah, I was on a different machine than normal. Thrashing resolved!

I'll add the hardcoded epochs for when rewards began.

File: explorer/src/providers/accounts/rewards.tsx (hunk 1 above)
Comment (oJshua) [reply explorer/src/providers/accounts/rewards.tsx]: The full response is something like 'Failed to fetch inflation reward: Block not available for...', which should no longer happen on my latest commit.

**Review by jstarry — APPROVED** [review]

**Pull request merged**